CHANGELOG.md

```diff
@@ -6,15 +6,43 @@ All Notable changes to `Period` will be documented in this file
 
 ### Added
 
-- None
+- Added support for the boundary types on the package
+    - `Period::EXCLUDE_START_INCLUDE_END`
+    - `Period::INCLUDE_START_EXCLUDE_END`
+    - `Period::EXCLUDE_ALL`
+    - `Period::INCLUDE_ALL`
+    - `Period::getBoundaryType`
+    - `Period::isStartExcluded`
+    - `Period::isStartIncluded`
+    - `Period::isEndExcluded`
+    - `Period::isEndIncluded`
+    - `Period::withBoundaryType`
+    - `Period::__construct` adds the `$boundaryType` argument;
+    - `Period::after` adds the `$boundaryType` argument;
+    - `Period::before` adds the `$boundaryType` argument;
+    - `Period::around` adds the `$boundaryType` argument;
+    - `Period::fromDatePeriod` adds the `$boundaryType` argument;
+- Added missing [Allen's Algebra intervals](https://www.ics.uci.edu/~alspaugh/cls/shr/allen.html)
+    - `Period::borderOnStart`
+    - `Period::borderOnEnd`
+    - `Period::isDuring`
+    - `Period::starts`
```

Comment:
For `Period::starts()` and `Period::ends()`, I think semantics can be improved, because I needed to reach the docs to know what `starts` and `ends` are for and what they do. 

What about `Period::hasSameStartingPoint()` and `Period::hasSameEndingPoint()` or something like this? 

I know it's a bit longer but I can understand at a glance the method's purpose and what they they return (a boolean), without having to look at the code or at the docs.

Reply:
I've added them because now that boundaries type are  taken into accout 2 periods with the same starting datepoint do not necessary starts at the same "time" if one include the starting point and the other does not 🤔 . So you suggested name would be even more missleading don't you think ?

Reply:
Maybe we could rename them `Period::begins` and `Period:finishes` 🤔

Reply:
Of note, to be fully compliant with Allen's proposal, I should add a `Period::beginnedBy` and a `Period::finishedBy` methods but I don't know if it's very crucial/important/useful since they just "reverse" `Period::begins` and a `Period::finishes`, thoughts ?

Reply:
TL; DR: `Period::startsBy()` and `Period::endsBy()` should do the trick 😅

> Of note, to be fully compliant with Allen's proposal, I should add a Period::beginnedBy and a Period::finishedBy methods but I don't know if it's very crucial/important/useful since they just "reverse" Period::begins and Period::finishes, thoughts ?

Since both methods return a boolean, _ideally_ they should start with a `is`, `has`, or any verb that expect a yes / no answer.

Besides, since these methods compare the current object to another one, they should end with a comparison operator, i.e. `by`, `like`, `as`, `than`, etc.

At first glance, when I read `Period::beginnedBy`, I know that the method will expect at least one argument, but I don't really know if it will _do_ or _return_ something. 

If it _does_ something I would name it `Period::beginBy()`, if it _returns_ a boolean I would name it `Period::beginsBy()` or `Period::isBeginnedBy()` (I must admit I prefer the 1st one).

Moreover, I think we should avoid using multiple words with the same meaning (begin / start, end / finish); since there's a `Period::getStartDate()` method, I would preferably expect a `Period::startsBy()` method.

> I've added them because now that boundaries type are taken into accout 2 periods with the same starting datepoint do not necessary starts at the same "time" if one include the starting point and the other does not 🤔 . So you suggested name would be even more missleading don't you think ?

The methods shouldn't be named with something related to a datepoint indeed, since they expect a `Period` object as an argument, and not a startingPoint / and endingPoint., so my `Period::hasSameStartingPoint()` and `Period::hasSameEndingPoint()` were not that relevant.

Reply:
done!!  🎉